sdk/tests/js/tests/compressed-tex-image.js

```diff
@@ -116,23 +118,25 @@ if (!gl) {
       wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION,
           `gl.compressedTexImage2D(gl.TEXTURE_2D, 0, ext.${enumName}, ${blockSize},${blockSize}, 0, ${blockByteSize}, ${blockByteSize+1})`);
 
-      wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
-          `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, 0)`);
-      wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
-          `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, 1)`);
-      wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
-          `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, ${blockByteSize})`);
-      wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION,
-          `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, ${blockByteSize+1})`);
+      if (supportsSubImage) {
+        wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
+            `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, 0)`);
+        wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
+            `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, 1)`);
+        wtu.shouldGenerateGLError(gl, gl.NO_ERROR,
+            `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, ${blockByteSize})`);
+        wtu.shouldGenerateGLError(gl, gl.INVALID_OPERATION,
+            `gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0,0, ${blockSize},${blockSize}, ext.${enumName}, ${blockByteSize}, ${blockByteSize+1})`);
+      }
```

Comment:
Please add in the negative tests here too.

Reply:
While WEBGL_compressed_texture_etc1 does not support compressedTexSubImage2D, WEBGL_compressed_texture_etc does support compressedTexSubImage2D, so negative tests would be conditional on having WEBGL_compressed_texture_etc1 and NOT WEBGL_compressed_texture_etc.

I'll add it, but it will add a bit of complexity to the test.

Reply:
Oh, never mind, WEBGL_compressed_texture_etc is exclusively for ETC2, so the 2 are unrelated